spark-bigquery-connector-common/src/main/java/com/google/cloud/spark/bigquery/BigQueryConnectorUtils.java

```diff
@@ -52,4 +57,32 @@ static SparkBigQueryPushdown createSparkBigQueryPushdown() {
             "Could not find an implementation of %s that supports Spark version %s",
             SparkBigQueryPushdown.class.getName(), sparkVersion));
   }
+
+  public static final class ProtobufSchemaFieldCacheEntry {
+    DataType sparkType;
+    boolean nullable;
+    Descriptors.Descriptor nestedTypeDescriptor;
+    Descriptors.FieldDescriptor fieldDescriptor;
+    Optional<TypeConverter> typeConverterOptional;
+    Optional<SupportedCustomDataType> customDataTypeOptional;
```

Comment:
Please use getters + (setters / constuctors).

Reply:
Also, this class should be moved to ProtobufUtils.

Reply:
Done